src/components/Grid/useDataGridSource.tsx

```diff
@@ -390,7 +390,10 @@ export const useDataGridSource = <
         ...(!isCacheComplete ? { apiSortModel: sortModel } : {}),
       }));
 
-      apiRef.current.scrollToIndexes({ rowIndex: 0 });
+      // ensures datagrid is mounted before scrolling
+      if (apiRef.current.rootElementRef.current) {
+        apiRef.current.scrollToIndexes({ rowIndex: 0 });
+      }
```

Comment:
Why? The the initial state `restore` is triggering `onSortModelChange`/`onFilterModelChange`?

I guess this could be a good thing to sync the internal state here with the current/initial view, right?
Just not for the scrolling side effect. Right?